diff --git a/Makefile b/Makefile
index 5a3bab7..0658014 100644
--- a/Makefile
+++ b/Makefile
@@ -482,7 +482,7 @@ else ifeq ($(platform), miyoo)
 	CXX = /opt/miyoo/usr/bin/arm-linux-g++
 	AR = /opt/miyoo/usr/bin/arm-linux-ar
 	SHARED := -shared -nostdlib -Wl,--version-script=link.T
-	fpic := -fPIC -DPIC
+	fpic := -fno-PIC
 	CFLAGS += -fomit-frame-pointer -ffast-math -march=armv5te -mtune=arm926ej-s
 	CFLAGS += -DSMALL_TRANSLATION_CACHE
 	HAVE_DYNAREC := 1
