diff --git a/libretro-build-common.sh b/libretro-build-common.sh
index cbe24739..9451b423 100644
--- a/libretro-build-common.sh
+++ b/libretro-build-common.sh
@@ -20,7 +20,7 @@ die() {
 #
 #  echo_cmd "$MAKE TARGET=\"libretro\" $COMPILER OTHERVAR=\"$SOMETHING\""
 #
-if [ "${CC}" ] && [ "${CXX}" ]; then
+if [ "${CC}" ] && [ "${CXX}" ] && [ "${CC}" != "gcc" ] && [ "${CXX}" != "g++" ]; then
 	COMPILER="CC=\"${CC}\" CXX=\"${CXX}\""
 else
 	COMPILER=""
@@ -445,6 +445,7 @@ summary() {
 		secho "$(color 31)$num_fail core(s)$(color) failed:"
 		lecho "$num_fail core(s) failed:"
 		lsecho "$fmt_fail"
+		exit 1
 	fi
 }
 
