diff --git a/Makefile b/Makefile
index b6af83f..d6d5d83 100644
--- a/Makefile
+++ b/Makefile
@@ -250,14 +250,19 @@ else ifeq ($(platform), miyoo)
    CC = /opt/miyoo/usr/bin/arm-linux-gcc
    CXX = /opt/miyoo/usr/bin/arm-linux-g++
    AR = /opt/miyoo/usr/bin/arm-linux-ar
-   fpic := -fPIC
+   fpic := 
    ARM_ASM = 1
    ASM_CPU = 0
    ASM_SPC700 = 0
    SHARED := -shared -Wl,--version-script=libretro/link.T -Wl,--no-undefined
-   CFLAGS += -fomit-frame-pointer -ffast-math -fno-unroll-loops -flto -mcpu=arm926ej-s
-   CFLAGS += -DFAST_ALIGNED_LSB_WORD_ACCESS
-
+   CFLAGS += -fomit-frame-pointer -ffast-math -flto -mcpu=arm926ej-s -Ofast
+   CFLAGS += -DFAST_ALIGNED_LSB_WORD_ACCESS \
+      -DDINGUX \
+      -DLSB_FIRST \
+      -DPSS_STYLE=1 \
+      -DHAVE_ASPRINTF \
+      -DFRAMESKIP \
+      -D_REENTRANT
 # Windows MSVC 2010 x64
 else ifeq ($(platform), windows_msvc2010_x64)
 	CC  = cl.exe
@@ -396,7 +401,7 @@ LIBRETRO_DIR := ./libretro
 ifeq ($(DEBUG), 1)
 DEFINES += -O0 -g
 else ifneq (,$(findstring msvc,$(platform)))
-DEFINES += -O2 -DNDEBUG=1
+DEFINES += -O3 -DNDEBUG=1
 else
 DEFINES += -O3 -DNDEBUG=1
 endif
diff --git a/libretro/libretro_core_options.h b/libretro/libretro_core_options.h
index ad03eae..455debb 100644
--- a/libretro/libretro_core_options.h
+++ b/libretro/libretro_core_options.h
@@ -68,7 +68,7 @@ struct retro_core_option_v2_definition option_defs_us[] = {
          { "fixed_interval", "Fixed Interval" },
          { NULL, NULL },
       },
-      "disabled"
+      "auto"
    },
    {
       "snes9x2002_frameskip_threshold",
@@ -133,7 +133,7 @@ struct retro_core_option_v2_definition option_defs_us[] = {
          { "enabled",  NULL },
          { NULL, NULL },
       },
-      "enabled"
+      "disabled"
    },
    {
       "snes9x2002_low_pass_filter",
