diff --git a/libavutil/mem.h b/libavutil/mem.h
index f9d88847..5d505d22 100644
--- a/libavutil/mem.h
+++ b/libavutil/mem.h
@@ -335,7 +335,7 @@ av_alloc_size(2, 3) void *av_realloc_array(void *ptr, size_t nmemb, size_t size)
  * @warning Unlike av_malloc(), the allocated memory is not guaranteed to be
  *          correctly aligned.
  */
-av_alloc_size(2, 3) int av_reallocp_array(void *ptr, size_t nmemb, size_t size);
+int av_reallocp_array(void *ptr, size_t nmemb, size_t size);

 /**
  * Reallocate the given buffer if it is not large enough, otherwise do nothing.
