From a5fdd63c17a1410c5a81c4d55faebfe45305550c Mon Sep 17 00:00:00 2001
From: Apaczer <94932128+Apaczer@users.noreply.github.com>
Date: Mon, 10 Mar 2025 20:39:21 +0100
Subject: [PATCH] Makefile.miyoo: use STAGING dir of BR2

---
 Makefile.miyoo | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/Makefile.miyoo b/Makefile.miyoo
index 2880af10ad..68e047898c 100644
--- a/Makefile.miyoo
+++ b/Makefile.miyoo
@@ -3,7 +3,7 @@
 #########################
 
 # Default toolchain directory
-TOOLCHAIN_DIR=/opt/miyoo
+TOOLCHAIN_DIR="$(STAGING_DIR)"
 
 # All toolchain-related variables may be
 # overridden via the command line
@@ -26,11 +26,11 @@ else
 STRIP                 = $(TOOLCHAIN_DIR)/usr/bin/arm-linux-strip
 endif
 
-GCW0_SDL_CONFIG      ?= $(TOOLCHAIN_DIR)/usr/arm-miyoo-linux-uclibcgnueabi/sysroot/usr/bin/sdl-config
-GCW0_FREETYPE_CONFIG ?= $(TOOLCHAIN_DIR)/usr/arm-miyoo-linux-uclibcgnueabi/sysroot/usr/bin/freetype-config
+GCW0_SDL_CONFIG      ?= $(TOOLCHAIN_DIR)/usr/bin/sdl-config
+GCW0_FREETYPE_CONFIG ?= $(TOOLCHAIN_DIR)/usr/bin/freetype-config
 
-GCW0_INC_DIR         ?= $(TOOLCHAIN_DIR)/usr/arm-miyoo-linux-uclibcgnueabi/sysroot/usr/include
-GCW0_LIB_DIR         ?= $(TOOLCHAIN_DIR)/usr/arm-miyoo-linux-uclibcgnueabi/sysroot/usr/lib
+GCW0_INC_DIR         ?= $(TOOLCHAIN_DIR)/usr/include
+GCW0_LIB_DIR         ?= $(TOOLCHAIN_DIR)/usr/lib
 
 #########################
 #########################
-- 
2.45.2.windows.1

