From b3000a0120290c36f6fbb11f87d1642796dbc58b Mon Sep 17 00:00:00 2001
From: Apaczer <94932128+Apaczer@users.noreply.github.com>
Date: Mon, 10 Mar 2025 20:43:32 +0100
Subject: [PATCH] config.def.h: reduce audio delay to 64

---
 config.def.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.def.h b/config.def.h
index 6b67763eae..673219ae5f 100644
--- a/config.def.h
+++ b/config.def.h
@@ -1165,7 +1165,7 @@
 
 /* Desired audio latency in milliseconds. Might not be honored
  * if driver can't provide given latency. */
-#if defined(ANDROID) || defined(EMSCRIPTEN) || defined(RETROFW) || defined(MIYOO)
+#if defined(ANDROID) || defined(EMSCRIPTEN) || defined(RETROFW)
 /* For most Android devices, 64ms is way too low. */
 #define DEFAULT_OUT_LATENCY 128
 #define DEFAULT_IN_LATENCY 128
-- 
2.45.2.windows.1

