From 0b53f1b9a799e20d3cd30f2da09f7a52b97dd5d3 Mon Sep 17 00:00:00 2001
From: Apaczer <94932128+Apaczer@users.noreply.github.com>
Date: Wed, 30 Apr 2025 23:04:15 +0200
Subject: [PATCH] configuration: set active "Unified Menu Controls"

---
 configuration.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configuration.c b/configuration.c
index b8fabf3021..d3aba25012 100644
--- a/configuration.c
+++ b/configuration.c
@@ -1950,7 +1950,11 @@ static struct config_bool_setting *populate_settings_bool(
 #endif
 
 #ifdef HAVE_MENU
+#ifdef MIYOO
+   SETTING_BOOL("menu_unified_controls",         &settings->bools.menu_unified_controls, true, true, false);
+#else
    SETTING_BOOL("menu_unified_controls",         &settings->bools.menu_unified_controls, true, false, false);
+#endif
    SETTING_BOOL("menu_disable_info_button",      &settings->bools.menu_disable_info_button, true, false, false);
    SETTING_BOOL("menu_disable_search_button",    &settings->bools.menu_disable_search_button, true, false, false);
    SETTING_BOOL("menu_disable_left_analog",      &settings->bools.menu_disable_left_analog, true, false, false);
-- 
2.45.2.windows.1

