From 7d4894a168d9bfc733ca194b87a8fde9365955ae Mon Sep 17 00:00:00 2001
From: Apaczer <94932128+Apaczer@users.noreply.github.com>
Date: Wed, 30 Apr 2025 23:04:15 +0200
Subject: [PATCH] configuration: set active "Unified Menu Controls"

---
 configuration.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configuration.c b/configuration.c
index 49bfb8fe1b..36c81fd041 100644
--- a/configuration.c
+++ b/configuration.c
@@ -1962,7 +1962,11 @@ static struct config_bool_setting *populate_settings_bool(
 #endif
 
 #ifdef HAVE_MENU
+#ifdef MIYOO
+   SETTING_BOOL("menu_unified_controls",         &settings->bools.menu_unified_controls, true, true, false);
+#else
    SETTING_BOOL("menu_unified_controls",         &settings->bools.menu_unified_controls, true, false, false);
+#endif
    SETTING_BOOL("menu_disable_info_button",      &settings->bools.menu_disable_info_button, true, false, false);
    SETTING_BOOL("menu_disable_search_button",    &settings->bools.menu_disable_search_button, true, false, false);
    SETTING_BOOL("menu_disable_left_analog",      &settings->bools.menu_disable_left_analog, true, false, false);
-- 
2.45.2.windows.1

