From 504fdd02ddb5504da0596b81d35810e9c13a0b0c Mon Sep 17 00:00:00 2001
From: Apaczer <94932128+Apaczer@users.noreply.github.com>
Date: Sat, 27 Dec 2025 23:21:47 +0100
Subject: [PATCH] nput_driver: fix toggle menu without hotkey press

---
 input/input_driver.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/input/input_driver.c b/input/input_driver.c
index 153c5d52e6..0b372b336f 100644
--- a/input/input_driver.c
+++ b/input/input_driver.c
@@ -6057,7 +6057,7 @@ static void input_keys_pressed(
    if (     !any_pressed
          && !(input_st->flags & INP_FLAG_WAIT_INPUT_RELEASE)
          && !(input_st->flags & INP_FLAG_KB_MAPPING_BLOCKED)
-         && binds[port][RARCH_MENU_TOGGLE].key == binds[port][RARCH_ENABLE_HOTKEY].key)
+         && binds[port][RARCH_MENU_TOGGLE].key)
    {
       i = RARCH_MENU_TOGGLE;
 
-- 
2.45.2.windows.1

