From 79428a587d2806695e022ab7a2c3179d993d3c45 Mon Sep 17 00:00:00 2001
From: tiopex <tiopxyz@gmail.com>
Date: Tue, 10 Jun 2025 13:41:23 +0200
Subject: [PATCH] Add cedar support

---
 omx/gstomxvideo.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/omx/gstomxvideo.c b/omx/gstomxvideo.c
index 9feda19..018686e 100644
--- a/omx/gstomxvideo.c
+++ b/omx/gstomxvideo.c
@@ -40,6 +40,9 @@ gst_omx_video_get_format_from_omx (OMX_COLOR_FORMATTYPE omx_colorformat)
   GstVideoFormat format;
 
   switch (omx_colorformat) {
+    case OMX_COLOR_FormatYVU420Planar:
+      format = GST_VIDEO_FORMAT_YV12;
+      break;
     case OMX_COLOR_FormatL8:
       format = GST_VIDEO_FORMAT_GRAY8;
       break;
-- 
2.34.1

