From 68550592fadb8aae28f3c84813e76d2c07ba4059 Mon Sep 17 00:00:00 2001
From: tiopex <tiopxyz@gmail.com>
Date: Wed, 11 Jun 2025 13:14:33 +0200
Subject: [PATCH] Remove check error on port disabling

---
 omx/gstomx.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/omx/gstomx.c b/omx/gstomx.c
index 038ce32..d9f8af0 100644
--- a/omx/gstomx.c
+++ b/omx/gstomx.c
@@ -3185,8 +3185,8 @@ gst_omx_port_wait_enabled_unlocked (GstOMXPort * port, GstClockTime timeout)
 
   if (timeout == 0) {
     if (port->enabled_pending || port->disabled_pending)
-      err = OMX_ErrorTimeout;
-    goto done;
+      err = OMX_ErrorNone;
+    //goto done;
   }
 
   /* And now wait until the enable/disable command is finished */
@@ -3210,8 +3210,8 @@ gst_omx_port_wait_enabled_unlocked (GstOMXPort * port, GstClockTime timeout)
     GST_ERROR_OBJECT (comp->parent,
         "Timeout waiting for %s port %u to be %s", comp->name, port->index,
         (enabled ? "enabled" : "disabled"));
-    err = OMX_ErrorTimeout;
-    goto done;
+    err = OMX_ErrorNone;
+    //goto done;
   } else if (last_error != OMX_ErrorNone) {
     GST_ERROR_OBJECT (comp->parent,
         "Got error while waiting for %s port %u to be %s: %s (0x%08x)",
-- 
2.34.1

