diff --git a/dingux/dingux_utils.c b/dingux/dingux_utils.c
index c6d5b791bf..7f2ca9ed92 100644
--- a/dingux/dingux_utils.c
+++ b/dingux/dingux_utils.c
@@ -37,7 +37,7 @@
 /* Base path defines */
 #define DINGUX_HOME_ENVAR                 "HOME"
 #define DINGUX_BASE_DIR                   "retroarch"
-#define DINGUX_BASE_DIR_HIDDEN            ".retroarch"
+#define DINGUX_BASE_DIR_HIDDEN            "retroarch"
 #define DINGUX_RS90_MEDIA_PATH            "/media"
 #define DINGUX_RS90_DEFAULT_SD_PATH       "/media/mmcblk0p1"
 #define DINGUX_RS90_DATA_PATH             "/media/data"
