diff --git a/rules.d/core-rules.sh b/rules.d/core-rules.sh
index f80055be..a355d27b 100644
--- a/rules.d/core-rules.sh
+++ b/rules.d/core-rules.sh
@@ -1121,7 +1121,7 @@ include_core_picodrive() {
 libretro_picodrive_name="Picodrive"
 libretro_picodrive_git_url="https://github.com/libretro/picodrive.git"
 libretro_picodrive_git_submodules="yes"
-libretro_picodrive_build_makefile="Makefile.libretro"
+libretro_picodrive_build_makefile="Makefile.libretro -f ../Makefile.libretro-picodrive"

 include_core_tgbdual() {
        register_module core "tgbdual" -ngc -ps3 -wii
@@ -1480,7 +1480,7 @@ libretro_tic80_git_submodules="yes"
 libretro_tic80_build_subdir="core"
 libretro_tic80_build_extradir="lib/"
 libretro_tic80_build_rule="cmake"
-libretro_tic80_build_args="-DBUILD_PLAYER=OFF -DBUILD_SOKOL=OFF -DBUILD_SDL=OFF -DBUILD_DEMO_CARTS=OFF -DBUILD_LIBRETRO=ON -DBUILD_WITH_MRUBY=OFF -DCMAKE_BUILD_TYPE=Release"
+libretro_tic80_build_args="-DBUILD_PLAYER=OFF -DBUILD_SOKOL=OFF -DBUILD_SDL=OFF -DBUILD_DEMO_CARTS=OFF -DBUILD_LIBRETRO=ON -DBUILD_WITH_MRUBY=OFF -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS=-DDINGUX"

 include_core_lowresnx() {
        register_module core "lowresnx" -theos_ios -ngc -sncps3 -ps3 -psp1 -qnx -wii